### PR TITLE
otp-ui itinerary body

### PR DESCRIPTION
**Pull request by evansiroky**
Replaces the itinerary-body with the otp-ui itinerary body. This PR is built on top of #140.

### CODE REVIEW

File: lib/components/mobile/results-screen.js [hunk 1]
```diff
@@ -95,24 +118,24 @@ class MobileResultsScreen extends Component {
     */
 
     const locationsSummary = (
-      <div style={{ position: 'fixed', top: 50, left: 0, right: 0, height: 50, paddingRight: 10 }}>
-        <Row className='locations-summary' style={{ padding: '4px 8px' }}>
-          <Col xs={8} sm={11} style={{ fontSize: '1.1em', lineHeight: '1.2em' }}>
+      <LocationSummaryContainer>
+        <LocationsSummaryRow className='locations-summary'>
+          <LocationsSummaryColFromTo xs={8} sm={11}>
             <div className='location'>
```
Comment (binh-dam-ibigroup) [inline lib/components/mobile/results-screen.js]: Should this be `styled.div` too?

File: lib/components/mobile/results-screen.js [hunk 2]
```diff
@@ -95,24 +118,24 @@ class MobileResultsScreen extends Component {
...
-              <LocationIcon type='from' /> { query.from ? query.from.name : '' }
+              <StyledLocationIcon type='from' /> { query.from ? query.from.name : '' }
             </div>
             <div className='location' style={{ marginTop: '2px' }}>
```
Comment (binh-dam-ibigroup) [inline lib/components/mobile/results-screen.js]: Should this be styled.div too?

File: lib/components/map/bounds-updating-overlay.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    if (oldProps.popupLocation || newProps.popupLocation) return
+
+    const { map } = newProps.leaflet
+    if (!map) return
+
+    const padding = [30, 30]
+
+    // Fit map to to entire itinerary if active itinerary bounds changed
+    const oldItinBounds = oldProps.itinerary && getItineraryBounds(oldProps.itinerary)
+    const fromChanged = !isEqual(oldProps.query && oldProps.query.from, newProps.query && newProps.query.from)
```
Comment (binh-dam-ibigroup) [inline lib/components/map/bounds-updating-overlay.js]: Suggestion: Create local variables such as
```js
const newFrom = newProps.query && newProps.query.from
const newTo = ...
const oldFrom = ...
const oldTo = ...
```

Comment (binh-dam-ibigroup) [reply lib/components/map/bounds-updating-overlay.js]: ... and replace where applicable.

**Comment by evansiroky** [comment]
@binh-dam-ibigroup, the comments you have made are regarding changes made to #140, so I pushed changes to that branch.

**Comment by landonreed** [comment]
It's a little tricky to isolate what commits are from this PR vs. #140. @evansiroky, now that 140 is approved, could you merge into dev and merge dev into this, so we can see the isolated commits? Or just provide a link to the commit range that is applicable here.

**Comment by evansiroky** [comment]
All up-to-date. Also fixed the map display in the print layout.

**Comment by evansiroky** [comment]
> There should be a mention that itinerary body now requires config.dateTime and config.transitOperators to be populated in config.yml? (Those are not required when using `DefaultItinerary`.)

Right, I should've mentioned the config changes. I'm using a slightly modified config file from trimet-mod-otp (I'm actually transpiling the code and copying it into trimet-mod-otp and running it all there). I have renamed the `operators` config key to `transitOperators` and have commented out all lines from the `alternateTransitIndex`.

File: lib/components/narrative/line-itin/connected-itinerary-body.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+          RouteDescription={RouteDescription}
+          setActiveLeg={setActiveLeg}
+          setLegDiagram={showLegDiagram}
+          setViewedTrip={setViewedTrip}
+          showAgencyInfo
+          showElevationProfile
+          showLegIcon
+          showMapButtonColumn={false}
+          showViewTripButton
+          toRouteAbbreviation={noop}
```
Comment (landonreed) [inline lib/components/narrative/line-itin/connected-itinerary-body.js:54]: Can we give this a comment? Not fully aware of why a noop func would need to be passed here, but that seems like a strange design.

File: lib/components/narrative/line-itin/connected-itinerary-body.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import { setActiveLeg } from '../../../actions/narrative'
+import { setViewedTrip } from '../../../actions/ui'
+import TransitLegSubheader from './connected-transit-leg-subheader'
+import TripDetails from '../connected-trip-details'
+import TripTools from '../trip-tools'
+
+const noop = () => {}
+
+class ConnectedItineraryBody extends Component {
+  shouldComponentUpdate (nextProps, nextState) {
```
Comment (landonreed) [inline lib/components/narrative/line-itin/connected-itinerary-body.js:21]: This shouldComponentUpdate deserves a comment about the conditions/user interactions it is trying to account for.

File: lib/components/narrative/line-itin/connected-itinerary-body.js (hunk 4 above)
Comment (binh-dam-ibigroup) [reply lib/components/narrative/line-itin/connected-itinerary-body.js:54]: It's the same dilemma between `callIfValid` or passing a noop by default...

Comment (landonreed) [reply lib/components/narrative/line-itin/connected-itinerary-body.js:54]: Can't we just provide a default prop?

Comment (evansiroky) [reply lib/components/narrative/line-itin/connected-itinerary-body.js:54]: See https://github.com/opentripplanner/otp-ui/pull/106.

**Comment by evansiroky** [comment]
Ready for review again, and relevant changes to other libraries have been proposed in the linked PRs.

**Comment by evansiroky** [comment]
@binh-dam-ibigroup, please assign me again when requesting changes on a PR. Thanks!

**Comment by binh-dam-ibigroup** [comment]
> @binh-dam-ibigroup, please assign me again when requesting changes on a PR. Thanks!

Ooops sorry.

**Comment by evansiroky** [comment]
I opted to replace the css with styled-components. Ready for review again.

**Comment by evansiroky** [comment]
> Approving, and opening issue #153 to deal with it separately.

Unassigning myself due to this.

**Pull request merged**